web/themes/custom/novel/templates/paragraphs/paragraph--recommendation.html.twig

```diff
@@ -0,0 +1,16 @@
+{% set application = react_app['#name'] %}
+
+{% if application == 'recommended-material' %}
```

Comment:
Instead of doing it like this, then why not make a hook suggestion, so you can have two files?

e.g.:

```
paragraph--recommendation.html.twig
```
and 
```
paragraph--recommendation--app.html.twig
```

(or whatever you wanna name it)

https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Render%21theme.api.php/function/hook_theme_suggestions_HOOK/8.5.x

Reply:
That way, Drupal also understands how to cache the paragraph more effectively :)

Reply:
Followed your suggestion. See changes in commit